### ISSUE TITLE
Loading songs via QWERTY and saving behavior seems changed.

### ISSUE THREAD

**Issue by ok-reza** [alter-alter/DelugeFirmware#14]
I have a lot of songs that start with a number eg. "10a.Song1", "10b.Song2", "11a.Song3", "12a.Song4", "12b.Song5"

Normally, when I load a song I'll type the first number on the keyboard. If I typed 1, the expected song to show up would be 12b.Song4: the song with the highest 2 digit number starting with 1 AND the highest letter would display. For some reason, this behavior is now changed so 10b.Song2 is chosen: the LOWEST 2 digit number starting with 1 with the HIGHEST letter after it.

I also noticed when overwriting a song, the behavior changes inconsistently. Say your song name is "HouseSong" Normally if you save the song "HouseSong_2" would automatically appear (and blink) as the new title before confirming the save. What happens now is "HouseSong_1" appears. Also, sometimes when canceling a save and saving again, "HouseSong_2" will show up instead of _1. 

Additionally, the 7SEG display should display and blink the end of the Song name's string: "g_2_" to alert the user that it is the 2nd iteration of the song and to make any edits to the title before saving. Instead, the beginning of the Song name's string "Hous" displays without blinking and you need to scroll to see the rest of the song name where it would reveal "_1" or sometimes "_2".

Lastly, If you hit LOAD then hit a letter (say X) on the keyboard that doesn't find a matching song, normally you would still see the letter "X_  " on display to finish typing what you want to load. With this firmware it won't let you type letters that don't match exisiting song names, instead it will jump to the nearest Song name based on the letter (or number). So it would reveal a project like "Vacuum" when you type X. Or it would show "12a.Song4" if you type the number "5" and had no songs with numbers between 12 and 5 on your SD card.

These issues seem related to #3. Perhaps whatever firmware version this fork started with already had inconsistencies between OLED and 7SEG?

**Comment by alter-alter** [comment]
I haven't made any changes to the saving process, so it's likely a modification made by the officials before the open-source release.

**Comment by sichtbeton** [comment]
Is the saving issue the same when building the official open source version?

**Comment by deltacrabmusic** [comment]
I'm having a similar issue with the latest 7seg firmware. Was using Deluge+firmware+V4p0p0-RC.bin with no issues, but now I'm finding that the loading behavior for kits has changed. I have some custom kits that I use frequently in a folder, labeled A1,A2,A3,..,,...C1,C2,C3,....H1,H2,... 

Now when I scroll through them, I get A1,C1,A2,C2,... Strangely, my H1 and H2 kit show up at the end when I scroll.

**Comment by alter-alter** [comment]
This firmware is simply built from the official source. Does it have similar problems?
[Deluge-release-7segOG.zip](https://github.com/alter-alter/DelugeFirmware/files/11783066/Deluge-release-7segOG.zip)

**Comment by ok-reza** [comment]
Yep! That seems to be the issue. Since the latest official Deluge 7SEG firmware is still behind the OLED one these changes haven't been realized yet.  Will have to further document the changes and figure out why these changes were made and decide how to move forward IMO.

Thanks alter-alter for providing the official bin.

**Comment by sichtbeton** [comment]
alter-alter is getting the deluge firmware GOAT fast.
